src/Evolution/Systems/Cce/Tags.hpp

```diff
@@ -0,0 +1,244 @@
+struct U : db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, 1>>;
+  static std::string name() noexcept { return "U"; }
+};
+
+/// Bondi parameter \f$\bar{U}\f$
+struct Ubar : db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, -1>>;
+  static std::string name() noexcept { return "Ubar"; }
+};
+
+/// Bondi parameter \f$U\f$
+struct W : db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, 0>>;
+  static std::string name() noexcept { return "W"; }
+};
+
+/// The derivative with respect to the numerical coordinate \f$y = 1 - 2R/r\f$
+template <typename Tag>
+struct Dy : db::PrefixTag, db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, Tag::type::type::spin>>;
+  using tag = Tag;
+  static const size_t dimension_to_differentiate = 2;
+  static const size_t dim = 3;
+  static std::string name() noexcept { return "Dy(" + Tag::name() + ")"; }
+};
+
+// prefix tags associated with the integrands which are used as input to solvers
+// for the CCE equations
+
+/// A prefix tag representing a quantity that will appear on the right-hand side
+/// of an explicitly regular differential equation
+template <typename Tag>
+struct Integrand : db::PrefixTag, db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, Tag::type::type::spin>>;
+  using tag = Tag;
+  static std::string name() noexcept {
+    return "Integrand(" + Tag::name() + ")";
+  }
+};
+
+/// A prefix tag representing the boundary data for a quantity on the extraction
+/// surface.
+template <typename Tag>
+struct BoundaryValue : db::PrefixTag, db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, Tag::type::type::spin>>;
+  using tag = Tag;
+  static std::string name() noexcept {
+    return "BoundaryValue(" + Tag::name() + ")";
+  }
+};
+
+/// A prefix tag representing the coefficient of a pole part of the right-hand
+/// side of a singular differential equation
+template <typename Tag>
+struct PoleOfIntegrand : db::PrefixTag, db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, Tag::type::type::spin>>;
+  using tag = Tag;
+  static std::string name() noexcept {
+    return "PoleOfIntegrand(" + Tag::name() + ")";
+  }
+};
+
+/// A prefix tag representing the regular part of the right-hand side of a
+/// regular differential equation
+template <typename Tag>
+struct RegularIntegrand : db::PrefixTag, db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, Tag::type::type::spin>>;
+  using tag = Tag;
+  static std::string name() noexcept {
+    return "RegularIntegrand(" + Tag::name() + ")";
+  }
+};
+
+/// A prefix tag representing a linear operator that acts on `Tag`. To determine
+/// the spin weight, It is assumed that the linear operator plays the role of
+/// \f$L\f$ in an equation of the form,
+/// \f$ (y - 1) \partial_y H + L H + L^\prime \bar{H} = A + (1 - y) B \f$
+template <typename Tag>
+struct LinearFactor : db::PrefixTag, db::SimpleTag {
```

Comment:
You use LinearFactor and LinearOperator interchangeably. I suggest to pick one. LinearOperator is actually quite general, so can you be more specific in naming this?

Reply:
I will revert all of them to 'linear factor'. I think the current version of the code should only have 'operator' in some comments. Ultimately, what I'm trying to communicate with this is a factor which acts in the precise way that is in the doxygen and `LinearFactor` seemed slightly better; do you have an alternative suggestion for a name for it?

Reply:
`LinearCoefficient` perhaps?

Reply:
After thinking about it, I disagree with `Coefficient`, as other parts of the code use it to refer to, for instance, spherical harmonic coefficients in a mode expansion.